src/fsharp/vs/service.fs

```diff
@@ -2289,10 +2286,7 @@ type BackgroundCompiler(legacyReferenceResolver, projectCacheSize, keepAssemblyC
     
 
     // STATIC ROOT: FSharpLanguageServiceTestable.FSharpChecker.parseFileInProjectCache. Most recently used cache for parsing files.
-    let parseFileInProjectCache = 
-        MruCache<ParseCacheLockToken, _, _>(parseFileInProjectCacheSize, 
-            areSame=AreSameForParsing3,
-            areSimilar=AreSubsumable3)
```

Comment:
Is it really ok to drop the ``areSimilar`` function here?

Reply:
@auduchinok Could you comment on the removal of ``areSimilar`` here?  Is it needed? Perhaps now ``same`` and ``similar`` are the same?

Reply:
I think I shouldn't have removed it completely. I've reverted and changed it to check file name. Please tell if it's a wrong change.

Reply:
That looks right